pkg/scanners/scannerv4/nodescan_convert.go

```diff
@@ -1,76 +1,93 @@
 package scannerv4
 
 import (
+	"fmt"
+	"regexp"
+	"strings"
+
 	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
 	"github.com/stackrox/rox/generated/storage"
-	"github.com/stackrox/rox/pkg/clair"
-	"github.com/stackrox/rox/pkg/cvss/cvssv3"
 	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+var (
+	rhcosOSImageRegexp = regexp.MustCompile(`(Red Hat Enterprise Linux) (CoreOS) ([\d])([\d]+)`)
 )
 
-func ToNodeScan(r *v4.VulnerabilityReport) *storage.NodeScan {
+func toNodeScan(r *v4.VulnerabilityReport, osImageRef string) *storage.NodeScan {
+	// TODO(ROX-26593): Instead of fixing labels here, add RHCOS DistributionScanner to ClairCore
+	fixedNotes := fixNotes(toStorageNotes(r.Notes), osImageRef)
+
 	return &storage.NodeScan{
-		ScanTime:       protocompat.TimestampNow(),
-		Components:     toStorageComponents(r),
-		Notes:          toStorageNotes(r.Notes),
-		ScannerVersion: storage.NodeScan_SCANNER_V4,
+		ScanTime:        protocompat.TimestampNow(),
+		Components:      toStorageComponents(r),
+		Notes:           fixedNotes,
+		ScannerVersion:  storage.NodeScan_SCANNER_V4,
+		OperatingSystem: toOperatingSystem(osImageRef),
```

Comment:
If `toOperatingSystem` returns `""`, then maybe we want to throw an error to the logs? Otherwise this may be difficult to debug